File: src/main/java/com/myimages/service/InviteService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.Invite;
import com.myimages.dto.InviteDTO;
import com.myimages.dto.InviteSearchDTO;
import com.myimages.dto.InvitePageDTO;
import com.myimages.dto.InviteConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface InviteService extends GenericService<Invite, Integer> {

	List<Invite> findAll();

	ResultDTO addInvite(InviteDTO inviteDTO, RequestDTO requestDTO);

	ResultDTO updateInvite(InviteDTO inviteDTO, RequestDTO requestDTO);

    Page<Invite> getAllInvites(Pageable pageable);

    Page<Invite> getAllInvites(Specification<Invite> spec, Pageable pageable);

	ResponseEntity<InvitePageDTO> getInvites(InviteSearchDTO inviteSearchDTO);
	
	List<InviteDTO> convertInvitesToInviteDTOs(List<Invite> invites, InviteConvertCriteriaDTO convertCriteria);

	InviteDTO getInviteDTOById(Integer inviteId);


}

File: myimages_ui/src/services/InviteService.js
import http from "../http-common"; 

class InviteService {
  getAllInvites(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/invite/invites`, searchDTO);
  }
 

  get(inviteId) {
    return this.getRequest(`/invite/${inviteId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/invite?field=${matchData}`, null);
  }

  addInvite(data) {
    return http.post("/invite/addInvite", data);
  }

  update(data) {
  	return http.post("/invite/updateInvite", data);
  }
  
  uploadImage(data,inviteId) {
  	return http.postForm("/invite/uploadImage/"+inviteId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new InviteService();
